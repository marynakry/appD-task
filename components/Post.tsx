import React, { Component } from 'react'
import styled from 'styled-components'

const Main = styled.div`
  font: 15px Helvetica, Arial;
  border: 1px solid #eee;
  padding: 0 10px;
`

const Title = styled.h1`
  font-size: 16px;
  font-weight: bold;
  margin: 10px 0;
`

interface PostProps {
  title: string
  postId: number
}

export class Post extends Component<PostProps> {
  render() {
    const { title, children, postId } = this.props
    return (
      <Main data-testid={`post-${postId}`}>
        <Title>{title}</Title>
        {children}
      </Main>
    )
  }
}
