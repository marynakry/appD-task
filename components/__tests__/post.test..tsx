import { afterEach, describe, expect, it } from '@jest/globals'
import { cleanup, getByTestId, render } from '@testing-library/react'
import { Post } from '../Post'

describe('Post', () => {
  afterEach(() => {
    cleanup()
  })

  it('render post component', () => {
    const { getByTestId } = render(
      <Post postId={1} title="title">
        test
      </Post>
    )
    const post = getByTestId('post-1')
    expect(post).toBeTruthy()
    expect(post).toMatchSnapshot()
    expect(post.textContent).toContain('test')
  })

  it('check post component title', () => {
    const { getByTestId } = render(
      <Post postId={1} title="title">
        test
      </Post>
    )
    const post = getByTestId('post-1')
    expect(post.getElementsByTagName('h1')).toHaveLength(1)
    expect(post.getElementsByTagName('h1')[0].textContent).toBe('title')
  })

  it('check post component empty title', () => {
    const { getByTestId } = render(
      <Post postId={1} title="">
        test
      </Post>
    )
    const post = getByTestId('post-1')
    expect(post.getElementsByTagName('h1')).toHaveLength(1)
    expect(post.getElementsByTagName('h1')[0].textContent).toBe('')
  })
})
