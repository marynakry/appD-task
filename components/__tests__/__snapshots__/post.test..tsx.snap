// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post render post component 1`] = `
<div
  class="Post__Main-sc-7gr9bq-0 jvyXjv"
  data-testid="post-1"
>
  <h1
    class="Post__Title-sc-7gr9bq-1 gEUbMS"
  >
    title
  </h1>
  test
</div>
`;
